objects/cash_register.rb

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env ruby
+
+class CashRegister
+  def total
+    @value = 0.0
+  end
+
+  def showing
+    @value
+  end
+
+  def purchase
+    @value = 3.50
+  end
+
+  def showing
+    @value
+  end
+
+  def pay
+    @value = 5.00 #=> "Your change is $1.50"
+  end
+
+  def showing
+    @value
+  end
+
+  def total
+    @value = 0.0
+  end
+
+  def showing
+    @value
+  end
+end
+
+
+
+register = CashRegister.new
```

Comment:
See the built_in_and_custom_objects.pptx presentation for example transactions.  Slides 17 and 18.